### ISSUE TITLE
Misleading Indentation Lint

### ISSUE THREAD

**Issue by programmerjake** [rust-lang/rust-clippy#6085]
### What it does

Checks for misleading indentation kinda like GCC's [`-Wmisleading-indentation`](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation).

### Categories (optional)

- Kind: style

Improves readability.

### Drawbacks

Can be annoying for generated code.

### Example

```rust
fn f() {
    if cond() {
    println!();}else{
    f2();            }
    f3();
}
```

Could be written as:

```rust
fn f() {
    if cond() {
        println!();
    } else {
        f2();
    }
    f3();
}
```

**Comment by ebroto** [comment]
Clippy is not well-equipped for this kind of lints because most of the time it works with an AST where whitespace is irrelevant. That said, I *think* this should be doable by checking the text of the relevant spans.

I'm not sure if the quality of the resulting lint will deserve a deny-by-default status.

@rust-lang/clippy should we add these kind of lints?

**Comment by Manishearth** [comment]
I would rather not. This is in the purview of `rustfmt`.

**Comment by oli-obk** [comment]
Yea, this is almost impossible to do in a reasonable way and after running rustfmt there's either no more misleading formatting or there's a bug in rustfmt which should be fixed in rustfmt and not papered over by clippy

**Comment by programmerjake** [comment]
> I would rather not. This is in the purview of `rustfmt`.

What about the people who don't use rustfmt?
